3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,234 @@
+//Item array removal
+
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+  ];
+ 
+  
+  // Write some code here
+  
+  const nameToRemove = names.splice(1,1);
+  // Code done
+  
+  console.log(names); // ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+
+//When will we be there??
+function calculateTravelTime(travelInformation) {
+    const { speed, destinationDistance } = travelInformation;
+  
+    // Calculate the total time in hours
+    const totalHours = destinationDistance / speed;
+  
+    // Convert totalHours to separate hours and minutes
+    const hours = Math.floor(totalHours);
+    const minutes = Math.round((totalHours - hours) * 60);
+  
+    // Format the result
+    const formattedTime = `${hours} hours and ${minutes} minutes`;
+  
+    return formattedTime;
+  }
+  
+  const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+  };
+  
+  const travelTime = calculateTravelTime(travelInformation);
+  console.log(travelTime); // Output: 8 hours and 38 minutes
+  
+  //Series duration of my life
+
+
+  const seriesDurations = [
+    {
+      title: "Game of Thrones",
+      days: 3,
+      hours: 1,
+      minutes: 0,
+    },
+    {
+      title: "Sopranos",
+      days: 3,
+      hours: 14,
+      minutes: 0,
+    },
+    {
+      title: "The Wire",
+      days: 2,
+      hours: 12,
+      minutes: 0,
+    },
+    // Add your favorite series here
+    {
+      title: "Dark",
+      days: 1,
+      hours: 10,
+      minutes: 0,
+    },
+  ];
+  
+  function calculateSeriesPercentage(seriesDurations) {
+    const averageLifespan = 80 * 365 * 24 * 60; // 80 years converted to minutes
+    let totalSeriesTime = 0;
+  
+    for (const series of seriesDurations) {
+      const seriesTimeMinutes =
+        series.days * 24 * 60 + series.hours * 60 + series.minutes;
+      totalSeriesTime += seriesTimeMinutes;
+    }
+  
+    const percentage = (totalSeriesTime / averageLifespan) * 100;
+  
+    return percentage.toFixed(3); // Format percentage to three decimal places
+  }
+  
+  function logOutSeriesText() {
+    for (const series of seriesDurations) {
+      const seriesPercentage = (
+        (series.days * 24 * 60 + series.hours * 60 + series.minutes) /
+        (80 * 365 * 24 * 60)
+      ).toFixed(3);
+      console.log(`${series.title} took ${seriesPercentage}% of my life`);
+    }
+  
+    const totalPercentage = calculateSeriesPercentage(seriesDurations);
+    console.log(`\nIn total, that is ${totalPercentage}% of my life`);
+  }
+  
+  logOutSeriesText();
+  
```

Comment:
👏 Fantastic job on this PR!

🌟 I appreciate the clarity in how you've approached the problem. Breaking down the main problem into smaller functions (like calculateSeriesPercentage) is a great practice, making the code more modular and readable.